### ISSUE TITLE
Confusing exception message when using ```cf update-service-broker```

### ISSUE THREAD

**Issue by JamesMustafa** [cloudfoundry/cloud_controller_ng#2504]
## Issue

When trying to update from one service broker to another using the ```cf update-service-broker``` command I got the following exception message:
```
Updating service broker (broker-name) as (my-username)...
Job (bfe405d8-952d-4bc5-a250-43952a4248d2) failed: An unknown error occurred.
FAILED
```

I tried to find more detailed information about the error and its cause in the logs but unfortunately nothing more useful was introduced there.

## Context

In order to update from one broker to other without making any changes to the offerings I used the same service/plan names in the catalog. I copied the service id as well but unfortunately forgot to replace the plan id from the old broker, which caused this error.

## Steps to Reproduce

1. Deploy a sample broker from [https://github.com/spring-cloud-samples/bookstore-service-broker](https://github.com/spring-cloud-samples/bookstore-service-broker)
2. Register the broker using ```cf create-service-broker``` command and enable the service plan.
3. Change the [plan Id](https://github.com/spring-cloud-samples/bookstore-service-broker/blob/main/src/main/java/org/springframework/cloud/sample/bookstore/servicebroker/config/ServiceCatalogConfiguration.java#L31) to some other guid.
4. Re-deploy the application
5. Use the ```cf update-service-broker``` command with the same parameters as the ```cf create-service-broker``` command in step 2.

## Expected result

More accurate exception title and message. For example: ```Plan name must be unique for the given plan id``` or just ```Plan name must be unique```

## Current result

```
Updating service broker (broker-name) as (my-username)...
Job (bfe405d8-952d-4bc5-a250-43952a4248d2) failed: An unknown error occurred.
FAILED
```

**Comment by philippthun** [comment]
Hi @JamesMustafa! On which CAPI version are you experiencing this error? When you call `api.cf.<your-foundation>/`, what version number is shown for `cloud_controller_v3`? I'm asking as there has been an improvement (PR #2443) with regards to error handling in CAPI 1.115.0 which resolves to `cloud_controller_v3.version = 3.104.0`.

**Comment by philippthun** [comment]
With PR #2510 the error message changes to:
```
Updating service broker (broker-name) as (my-username)...
Job (31bc193d-c769-436e-9e24-ec3c7f54b5f3) failed: Plan names must be unique within a service. Service bookstore already has a plan named standard
FAILED
```

**Comment by JamesMustafa** [comment]
Hi @philippthun, thanks for your support :)

**Comment by philippthun** [comment]
Reopening as the possible fix (#2537) is not yet available.

**Comment by sweinstein22** [comment]
Hi @JamesMustafa ,

We've pushed a fix that should remedy this issue. It'll go out as part of the next release, at that point if upgrading does not fix things feel free to reopen this issue. Thanks!
